vortex-array/src/array/byte_bool/compute/mod.rs

```diff
@@ -0,0 +1,242 @@
+use std::ops::{BitAnd, BitOr, BitXor, Not};
+use std::sync::Arc;
+
+use arrow_array::{ArrayRef as ArrowArrayRef, BooleanArray as ArrowBoolArray};
+use arrow_buffer::{BooleanBuffer, BooleanBufferBuilder};
+use num_traits::AsPrimitive;
+use vortex_dtype::{match_each_integer_ptype, Nullability};
+use vortex_error::{vortex_bail, VortexResult};
+use vortex_expr::Operator;
+use vortex_scalar::Scalar;
+
+use super::{ByteBoolArray, ByteBoolMetadata};
+use crate::ToArrayData;
+use crate::{
+    compute::{
+        as_arrow::AsArrowArray, compare::CompareFn, fill::FillForwardFn, scalar_at::ScalarAtFn,
+        slice::SliceFn, take::TakeFn, ArrayCompute,
+    },
+    encoding::ArrayEncodingRef,
+    stats::StatsSet,
+    validity::ArrayValidity,
+    ArrayDType, ArrayData, ArrayTrait, IntoArray,
+};
+
+impl ArrayCompute for ByteBoolArray {
+    fn as_arrow(&self) -> Option<&dyn crate::compute::as_arrow::AsArrowArray> {
+        Some(self)
+    }
+
+    fn compare(&self) -> Option<&dyn crate::compute::compare::CompareFn> {
+        Some(self)
+    }
+
+    fn fill_forward(&self) -> Option<&dyn crate::compute::fill::FillForwardFn> {
+        None
+    }
+
+    fn scalar_at(&self) -> Option<&dyn ScalarAtFn> {
+        Some(self)
+    }
+
+    fn slice(&self) -> Option<&dyn crate::compute::slice::SliceFn> {
+        Some(self)
+    }
+
+    fn take(&self) -> Option<&dyn crate::compute::take::TakeFn> {
+        Some(self)
+    }
+}
+
+impl ScalarAtFn for ByteBoolArray {
+    fn scalar_at(&self, index: usize) -> VortexResult<Scalar> {
+        if index >= self.len() {
+            vortex_bail!(OutOfBounds: index, 0, self.len());
+        }
+
+        let scalar = match self.is_valid(index).then(|| self.buffer()[index] == 1) {
+            Some(b) => Scalar::new(
+                vortex_dtype::DType::Bool(self.validity().nullability()),
+                vortex_scalar::ScalarValue::Bool(b),
+            ),
+            None => Scalar::null(self.dtype().clone()),
+        };
+
+        Ok(scalar)
+    }
+}
+
+impl AsArrowArray for ByteBoolArray {
+    fn as_arrow(&self) -> VortexResult<ArrowArrayRef> {
+        let nulls = self.logical_validity().to_null_buffer()?;
+
+        let mut builder = BooleanBufferBuilder::new(self.len());
+        builder.append_slice(self.as_ref());
```

Comment:
This AsRef feels dangerous. The problem is validity/nulls. It makes it too easy to forget about them. Would rather it be an explicit accessor like `self.maybe_null_slice()`. We should rename `PrimitiveArray.typed_data` to `PrimitiveArray.maybe_null_slice()` too!

Reply:
In your mind, should `maybe_null_slice` should still return `&[bool]`? I can also just make it a private method, so users won't be able to shoot themselves in the foot.

Reply:
Keep it public, users might need to access the bools themselves, but `pub fn maybe_null_slice(&self) -> &[bool]`

I merged the rename from `typed_data -> maybe_null_slice(&self) -> &[T]`